Assets/Scripts/HPPickupInteract.cs

```diff
@@ -0,0 +1,20 @@
+using UnityEngine;
+
+public class HPPickupInteract : MonoBehaviour
+{
+    // Private Variables \\
+    [SerializeField]
+    private int hpPickupVal { get; set; } = 100;
+
+    // Methods \\
+    public void OnTriggerEnter2D(Collider2D coll2D)
+    {
+        PlayerController player = coll2D.GetComponent<PlayerController>();
+
+        if (player.getCurrentHealth() != player.getMaxHealth())
```

Comment:
This will cause a `NullReferenceException` if the collider doesn't have a `PlayerController` component. You need to add a null check:
`if (player != null && player.getCurrentHealth() != player.getMaxHealth())`